### ISSUE TITLE
Investigate Theme Configuration Simplifications

### ISSUE THREAD

**Issue by jdbrannon** [react-bedrock/react-bedrock#17]
I want to answer some questions as an outcome to this

1. Can we simplify our Map usage? https://itnext.io/advanced-use-of-sass-maps-bd5a47ca0d1a
2. Should we create global CSS variables from the theme and reference those in our components instead of the map to reduce CSS?
3. Is there some automated setup that we can do to reduce the duplication of creating variable names through our components map and CSS classes that reference them from our component modules?
4. Can we create some sort of typing system when creating and consuming our theme objects to avoid typos?
5. What is the best way of iterating through our SCSS config maps and creating a fully expanded or expandable config document automatically for documentation purposes?

**Comment by jdbrannon** [comment]
4. I think a good option for this would be a VSCode extenstion. 

If we would like to create a VSCode extension to provide suggestions/autocomplete for the theme object and linting errors for an invalid object, one good source to investigate further in the future is Tailwind
https://github.com/tailwindlabs/tailwindcss-intellisense/tree/main
https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss

**Comment by jdbrannon** [comment]
5. We can probably just create a small node script that can run on build. I think we can move forward with that plan and once we create a task for taking care of that, more investigation can be done during that process if it seems necessary

**Comment by jdbrannon** [comment]
1. For now we are going to stick with the default map.get() method. The abstraction doesn't seem to give enough benefit and it will probably be easier for someone to understand the basic map.get() method by looking at the SCSS documentation

**Comment by jdbrannon** [comment]
2. Global CSS variables for the theme would be great. This will show where the values are truly originating from and theoretically would limit that max size of the css bundle because there would be more references to variables than to direct css. And the variables will have a smaller maximum size than the css

**Comment by jdbrannon** [comment]
I don't have all of the details completely fleshed out yet, but I think I have a good overview gameplan for moving forward:

We use json files as our source of truth for default themes. We can use an npm package called `json-to-scss` to convert the json files to scss maps as needed. And this will make it easier to parse for our `react-bedrock-intellisense` project and also for displaying default style objects in our styleguide without having to maintain it in more than one spot.

We already have some SCSS functions created to convert the same SCSS maps to CSS variables for use throughout the application. Using the structure of the json we let `react-bedrock-intellisense` handle the type safety.

Using json also means that we can make it easier on the end user to configure custom theming by just adding their own json config file. TODO: still need to figure out how we can use the npm package to convert the json to scss on build or any time that the file changes...

To recap answers to the questions being investigated:
1. We will use default SCSS functions and utilities whenever possible
2. We are going to use a custom SCSS function to automatically create the global CSS variables from our SCSS map
3. Similar to number 2, we still have one source of truth the json that gets converted to an SCSS map that creates CSS variables. `react-bedrock-intellisense` will have the typesafety generated from the json whenever we are referencing the CSS variables throughout both the creation of the NPM package and for the end user
4. We are creating a VSCode extension to add typesafety for the classnames
5. Now that we are using JSON as the source of truth, we can just use that for our config document
